src/call_hook.cpp

```diff
@@ -0,0 +1,62 @@
+#include "common.hpp"
+#include "call_hook.hpp"
+#include "memory/handle.hpp"
+
+namespace
+{
+	big::call_hook_memory g_call_hook_memory;
+}
+
+// https://github.com/martonp96/ClosedIV/blob/master/src/utils/memory.h#L64
+
+namespace big
+{
+	call_hook_memory::call_hook_memory()
+	{ 
+		m_memory = (uint8_t*)VirtualAlloc((void*)((uintptr_t)GetModuleHandle(0) + 0x20000000), 1024, MEM_COMMIT | MEM_RESERVE, PAGE_EXECUTE_READWRITE);
+		m_offset = 0;
+	}
+
+	call_hook_memory::~call_hook_memory()
+	{
+		VirtualFree(m_memory, 0, MEM_RELEASE);
+	}
+
+	void* call_hook_memory::allocate_jump_sequence(void* func)
+	{
+		m_offset = m_offset + ((16 - (m_offset % 16)) % 16); // align
+
+		*(int16_t*)&m_memory[m_offset] = 0xB848;
+		*(uint64_t*)&m_memory[m_offset+2] = (uint64_t)func;
+		*(int16_t*)&m_memory[m_offset+10] = 0xE0FF;

```

Comment:
Can't this be made into a class? You're declaring the char* member of m_memory, when you could instead declare the members and have the compiler figure it out. I don't see why all this array shifting math nonsense needs to occur.